### PR TITLE
Stat scaling

### PR DESCRIPTION
**Pull request by PBWiseman**
## Describe your changes

This branch is implementing stat scaling, so we have more ways to fine tune balancing of base values, items, and weapons. It also implements a basic version of a rocket launcher, though the exploding itself won't be done in this branch. It also has a VERY basic sword to start implementing that.

## Issue number and link

#65 

## Checklist before requesting a review
- [x] I have performed a self-review of my code
- [x] I have commented my code, particularly in hard-to-understand areas

### CODE REVIEW

**Comment by PBWiseman** [comment]
Made the stat and item description changes. This branch doesn't have the new items in it. It does have a very basic version of a rocket launcher and a sword in just to start testing them out. There are a lot of different changes in this branch, so I probably missed some errors. I spent a while testing and fixed some but please thoroughly test it for other issues.

**Comment by PBWiseman** [comment]
I have also added an excess crit chance to crit damage scaling at a 50% ratio. Not sure what ratio would be good, but I just wanted to get that in there.

**Review by AlexReidNZ — CHANGES_REQUESTED** [review]
The regen tick seems to happen way too fast, other than that the rest of this is ready for merging.

File: Assets/Scripts/Player/PlayerStats.cs [hunk 1]
```diff
@@ -94,16 +147,15 @@ void Update()
...
-        if (regenTick <= 0 && regenTrue && currentHealth < maxHealth) //only works if the player is missing health
+        if (nextRegenerationTick <= 0) //Health property will deal with the max health cap
         {
-            regenTick = regenInterval;
-            CurrentHealth += regenAmount;
-            Debug.Log($"Regen: {currentHealth}");
+            nextRegenerationTick = RegenerationDelay;
+            CurrentHealth += Math.Max(((MaxHealth * RegenerationPercentage) / 100), 1); //Regenerate the % of max health per tick. Minimum 1
         }
+        nextRegenerationTick -= Time.fixedDeltaTime;
```
Comment (AlexReidNZ) [inline Assets/Scripts/Player/PlayerStats.cs:158]: Regen ticks way too fast. I tried changing fixeddeltatime here back to deltatime and that seemed to fix it.

Comment (PBWiseman) [reply Assets/Scripts/Player/PlayerStats.cs]: Fixed

**Review by AlexReidNZ — APPROVED** [review]
Requested change has been made, all good for merge.

**Review by kblack89 — APPROVED** [review]
This code looks good and works with the intended stat scaling. No big errors from testing.

**Review by Loerad — APPROVED** [review]
Nothing jumps out as an issue. Looks good 👍

**Pull request merged**